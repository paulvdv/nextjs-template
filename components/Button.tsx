export default function Button({
  text,
  onClick = () => {},
}: React.PropsWithChildren<{
  text: string;
  onClick?: () => void;
}>): JSX.Element {
  return (
    <button className='btn' onClick={onClick}>
      {text}
    </button>
  );
}
