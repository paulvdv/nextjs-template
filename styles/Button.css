.btn {
  @apply flex h-10 items-center justify-center gap-2 rounded-full border border-solid px-4 text-sm sm:h-12 sm:px-5 sm:text-base;
}

.btn-primary {
  @apply border-transparent bg-foreground text-background transition-colors hover:bg-[#383838] dark:hover:bg-[#ccc];
}
