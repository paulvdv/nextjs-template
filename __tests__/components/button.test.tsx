import "@testing-library/jest-dom";
import { render, RenderResult, screen } from "@testing-library/react";
import Button from "../../components/Button";

describe("Button", () => {
  let tree: RenderResult;
  let clicked: Boolean;

  const text = "TESTButton";

  beforeEach(() => {
    clicked = false;
    tree = render(
      <Button
        text={text}
        onClick={() => {
          clicked = true;
        }}
      />,
    );
  });

  it("renders without changes", () => {
    expect(tree.container).toMatchSnapshot();
  });

  it("renders a button with prop text", () => {
    const button = screen.getByText(text);
    expect(button).toBeInTheDocument();
  });

  it("call the onClick callback when clicking the button", () => {
    const button = screen.getByText(text);

    button.click();

    expect(clicked).toBe(true);
  });
});
