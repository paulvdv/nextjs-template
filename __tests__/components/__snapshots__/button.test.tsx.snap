// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders without changes 1`] = `
<div>
  <button
    class="btn btn-primary"
  >
    TESTButton
  </button>
</div>
`;
